stake-pool/single-pool-cli/src/cli.rs

```diff
@@ -0,0 +1,399 @@
+    #[clap(long = "vote-account", value_parser = Pubkey::from_str)]
+    pub vote_account_address: Option<Pubkey>,
+
+    /// Signing authority on the stake account to be deposited. Defaults to the client keypair
+    #[clap(long = "stake-authority", id = "STAKE_WITHDRAW_AUTHORITY_KEYPAIR", validator = |s| is_valid_signer(s))]
+    pub stake_withdraw_authority: Option<SignerArg>,
+
+    /// The token account to mint to. Defaults to the client keypair's associated token account
+    #[clap(long = "token-account", value_parser = Pubkey::from_str)]
+    pub token_account_address: Option<Pubkey>,
+
+    /// The wallet to refund stake account rent to. Defaults to the client keypair's pubkey
+    #[clap(long = "lamports-to", value_parser = Pubkey::from_str)]
+    pub lamport_recipient_address: Option<Pubkey>,
+}
+
+#[derive(Clone, Debug, Args)]
+#[clap(group(ArgGroup::new("withdraw-amount").required(true).args(&["token-amount", "full-withdraw"])))]
+#[clap(group(ArgGroup::new("pool-source").required(true).args(&["pool-address", "vote-account-address"])))]
+// TODO undelegate flag, merge flag?
+pub struct WithdrawCli {
+    /// Amount of tokens to burn for withdrawal
+    pub token_amount: Option<u64>,
+
+    /// Instead of specifying a token amount, withdraw the maximum
+    #[clap(long = "full")]
+    pub full_withdraw: bool,
+
+    /// The token account to withdraw from. Defaults to the associated token account for the pool mint
+    #[clap(long = "token-account", value_parser = Pubkey::from_str)]
+    pub token_account_address: Option<Pubkey>,
+
+    /// The pool to withdraw from
+    #[clap(short, long = "pool", value_parser = Pubkey::from_str)]
+    pub pool_address: Option<Pubkey>,
+
+    /// The vote account corresponding to the pool to withdraw from
+    #[clap(long = "vote-account", value_parser = Pubkey::from_str)]
+    pub vote_account_address: Option<Pubkey>,
+
+    /// Signing authority on the token account. Defaults to the client keypair
+    #[clap(long = "token-authority", id = "TOKEN_AUTHORITY_KEYPAIR", validator = |s| is_valid_signer(s))]
+    pub token_authority: Option<SignerArg>,
+
+    /// Authority to assign to the new stake account. Defaults to the pubkey of the client keypair
+    #[clap(long = "stake-authority", value_parser = Pubkey::from_str)]
+    pub stake_authority_address: Option<Pubkey>,
+}
+
+#[derive(Clone, Debug, Args)]
+#[clap(group(ArgGroup::new("pool-source").required(true).args(&["pool-address", "vote-account-address"])))]
+pub struct CreateMetadataCli {
+    /// The pool to create default MPL token metadata for
+    #[clap(short, long = "pool", value_parser = Pubkey::from_str)]
+    pub pool_address: Option<Pubkey>,
+
+    /// The vote account corresponding to the pool to create metadata for
+    #[clap(long = "vote-account", value_parser = Pubkey::from_str)]
+    pub vote_account_address: Option<Pubkey>,
+}
+
+// XXX i didnt include uri because it looks useless for fungible tokens
```

Comment:
Isn't that where they get the token image from? Have I been misunderstanding how token metadata works this whole time?

Reply:
oh i have no idea actually. i was reading mpl docs and they just say stuff like "some external URI that can be anything" or "URI pointing to JSON representing the asset." i could add it back as an optional param

Reply:
Most of the time it's a pic or a video yep! But technically can be any string, as long as whatever client/UI is rendering it knows what to expecting